src/routes.ts

```diff
@@ -61,6 +62,24 @@ export default function(server: Hapi.Server) {
   });
   cacheControllers.push(cloudfrontController);
 
+  const incapsulaController = new IncapsulaController();
+  server.route({
+    method: 'DELETE',
+    path: '/v1/caches/incapsula/zones/{zone_id}',
+    options: incapsulaController.flushZoneConfig,
+  });
+  server.route({
+    method: 'DELETE',
+    path: '/v1/caches/incapsula/zones/{zone_id}/urls',
+    options: incapsulaController.flushURLsConfig,
+  });
+  server.route({
+    method: 'DELETE',
+    path: '/v1/caches/incapsula/zones/hostname',
```

Comment:
To be consistent, we should keep `{zone_id}` in the route

Reply:
For this 3rd route, I don't get why we should add the `zone_id` as this one is for flushing hostnames

Reply:
For consistency.
If another CDN propose such an API, it might require a `zone_id` (for example, we did not have implemented for Fasterize, but flushing by domain require the `zone_id`) and Darkstar should expose its own consistent API between CDN (ie. hide different concepts (zone, distribution, ...) between CDN).

For Fasterize, this is not a problem as we associate the CDN `zone_id` to a domain/hostname.

Reply:
Precisely, but in the case of Incapsula, hostnames are not associate with a zone_id.
So should the route be `/v1/caches/incapsula/zones/{zone_id}/hostname` where the `{zone_id}` is the actual hostname we want to flush

Reply:
I think so. 
Consistency is regarding the exposed routes by Darkstar.
If we have:
- `/v1/caches/fasterize/zones/{zone_id}/hostname`
- `/v1/caches/incapsula/zones/hostname`

We are not consistent any more and this is disturbing.

Moreover I suggest to replace `hostname` by `domain` into the Darkstar API. @abarre WDYT?

Reply:
There is a misunderstanding here.
There is no incapsula API to flush assets by hostname.

The only thing mentioning hostname is this part in the doc related to internal DNS cache :

Purge hostname from cache
Use this operation to purge the hostname from the cache.

This API is for customers who use the same CNAME provided by Imperva for multiple hostnames and would like to change the CNAME for a particular hostname. Purging the hostname is required for the CNAME change to take effect.

/api/prov/v1/sites/hostname/purge

So, unfortunatelly, I suggest to remove this route.

Reply:
👍 

> I am pretty sure that no real test as been done yet.
> We definitely need to validate the API with real token. @stefounet do you have access to Incapsula account ? Otherwise, we could ask the incapsula support to give a test account.

we've discussed that with @dax and agreed to add a apiKey to customer config that we'll use to call the Incapsula API

Reply:
but in the mean time, yeah, that'd be great to be able to test it if that's never been done before